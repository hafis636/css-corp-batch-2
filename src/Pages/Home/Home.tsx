import Product from '@components/Product';
import React, { memo, useEffect } from 'react';
import { ProductType } from 'types/productsTypes';

type Props = {
  products: ProductType[];
  loadProducts: () => void;
  loadCart: () => void;
  loading: any;
};

const Home = ({ products, loadProducts, loadCart, loading }: Props) => {
  useEffect(() => {
    loadProducts();
    loadCart();
  }, [loadProducts, loadCart]);

  return (
    <div className="bg-white">
      <div className="max-w-2xl mx-auto py-16 px-4 sm:py-24 sm:px-6 lg:max-w-7xl lg:px-8">

        <div className="mt-6 grid grid-cols-1 gap-y-10 gap-x-6 sm:grid-cols-2 lg:grid-cols-4 xl:gap-x-8">
          {loading['LOAD_PRODUCTS'] && (
            <div className="flex justify-center items-center text-white text-4xl bg-gray-400 absolute z-10 opacity-30">
              Loading...
            </div>
          )}
          {products?.map((product) => {
            return <Product key={product.id} {...product} />;
          })}
        </div>
      </div>
    </div>
  );
};

export default memo(Home);
