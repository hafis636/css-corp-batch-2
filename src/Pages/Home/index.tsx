import React from 'react';

interface Props {}

const Home = (props: Props) => {
  return (
    <div>
      <h1>Hello From Home</h1>
    </div>
  );
};

export default Home;
