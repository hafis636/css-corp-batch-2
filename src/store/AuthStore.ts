import { FormikHelpers } from 'formik';
import { makeAutoObservable } from 'mobx';
import { LoginInitValuesType } from 'Pages/Login/loginUtils';
import { RegisterInitValuesType } from 'Pages/Register/registerUtils';
import AuthService from 'services/AuthService';
import { User } from 'types/UserType';
import RootStore from '.';

export default class AuthStore {
  private authenticated = false;

  private user: User | undefined = undefined;

  rootStore: RootStore;

  constructor(rootStore: RootStore, private readonly authService: AuthService) {
    makeAutoObservable(this, { rootStore: false });
    this.rootStore = rootStore;
    this.authenticated = !!this.hasToken();
  }

  private setAuthenticated = (authenticated: boolean) => {
    this.authenticated = authenticated;
  };

  private setUser = (user?: User) => {
    this.user = user;
  };

  private hasToken = () => {
    return sessionStorage.getItem('@app/token');
  };

  onLogin = async (
    values: LoginInitValuesType,
    actions: FormikHelpers<LoginInitValuesType>,
  ) => {
    try {
      // this.rootStore.loadingStore.setLoading()
      const res = await this.authService.login(values);
      sessionStorage.setItem('@app/token', res.accessToken);
      actions.resetForm();
      this.setUser(res.user);
      this.setAuthenticated(true);
    } catch (error) {
      let message = 'Something went wrong. Please try after sometime.';
      if (error instanceof Error) {
        message = error.message;
      }
      actions.setErrors({ serverError: message });
    }
  };

  onRegister = async (
    values: RegisterInitValuesType,
    actions: FormikHelpers<RegisterInitValuesType>,
  ) => {
    try {
      const res = await this.authService.register(values);
      sessionStorage.setItem('@app/token', res.accessToken);
      actions.resetForm();
      this.setUser(res.user);
      this.setAuthenticated(true);
    } catch (error) {
      let message = 'Something went wrong. Please try after sometime.';
      if (error instanceof Error) {
        message = error.message;
      }
      actions.setErrors({ serverError: message });
    }
  };

  onLogout = async () => {
    sessionStorage.removeItem('@app/token');
    this.setAuthenticated(false);
    this.setUser();
  };

  isAuthenticated = () => {
    return this.authenticated;
  };

  getUser = () => {
    return this.user;
  };
}
