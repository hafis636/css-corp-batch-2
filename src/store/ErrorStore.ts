import { makeAutoObservable } from 'mobx';
import RootStore from '.';

export default class ErrorStore {
  rootStore: RootStore;

  constructor(rootStore: RootStore) {
    makeAutoObservable(this, { rootStore: false });
    this.rootStore = rootStore;
  }

  setError = () => {};

  hasError = (type: string) => {};
}
