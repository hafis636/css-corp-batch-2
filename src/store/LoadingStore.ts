import { makeAutoObservable } from 'mobx';
import RootStore from '.';

export default class LoadingStore {
  rootStore: RootStore;

  constructor(rootStore: RootStore) {
    makeAutoObservable(this, { rootStore: false });
    this.rootStore = rootStore;
  }

  setLoading = () => {};

  isLoading = (type: string) => {};
}
