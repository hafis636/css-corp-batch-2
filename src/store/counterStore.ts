import { action, flow, makeObservable, observable } from 'mobx';

class CounterStore {
  counter = 0;

  constructor() {
    makeObservable(this, {
      counter: observable,
      increment: action,
      decrement: action,
      loadCounter: flow,
    });
    // makeAutoObservable(this);
  }

  increment = () => {
    console.log('increment');

    this.counter++;
  };

  decrement = () => {
    console.log('decrement');
    this.counter--;
  };

  *loadCounter() {
    const res = yield axiosInstance.get('value');
    this.counter = res.data.val;
  }
}

export default CounterStore;
