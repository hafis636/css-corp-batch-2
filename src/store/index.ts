import AuthStore from './AuthStore';
import ErrorStore from './ErrorStore';
import LoadingStore from './LoadingStore';
import AuthService from '../services/AuthService';

export default class RootStore {
  authStore: AuthStore;
  errorStore: ErrorStore;
  loadingStore: LoadingStore;

  constructor() {
    this.authStore = new AuthStore(this, new AuthService());
    this.errorStore = new ErrorStore(this);
    this.loadingStore = new LoadingStore(this);
  }
}
