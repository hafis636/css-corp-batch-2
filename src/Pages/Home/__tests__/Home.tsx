import React from 'react';
import { render, screen } from '@testing-library/react';
import Home from '../index';
import { Provider } from 'react-redux';
import MockAdapter from 'axios-mock-adapter';
import store from '../../../configureStore';
import axiosInstance from 'utils/axios';

const mock = new MockAdapter(axiosInstance);

// const mockedAxios = axiosInstance as jest.Mocked<typeof axiosInstance>;

const setup = ({ ...props }) => {
  return render(
    <Provider store={store}>
      <Home />
    </Provider>,
  );
};

describe('Test Home page cases', () => {
  afterAll(() => {
    jest.clearAllMocks();
  });

  test('should take snapshot', () => {
    const { container } = setup({});
    expect(container.firstChild).toMatchSnapshot();
  });

  test('should take snapshot of product', async () => {
    mock.onGet('660/products').reply(200, [
      {
        id: 1,
        title: 'Fjallraven - Foldsack No. 1 Backpack, Fits 15 Laptops',
        price: 109.95,
        description:
          'Your perfect pack for everyday use and walks in the forest. Stash your laptop (up to 15 inches) in the padded sleeve, your everyday',
        category: "men's clothing",
        image: 'https://fakestoreapi.com/img/81fPKd-2AYL._AC_SL1500_.jpg',
        rating: {
          rate: 3.9,
          count: 120,
        },
      },
    ]);
    mock.onGet('660/cart').reply(200, [
      {
        productId: 1,
        quantity: 1,
        id: 2,
      },
    ]);
    const { container, findAllByTestId } = setup({});
    const products = await findAllByTestId('productContainer');
    expect(container.firstChild).toMatchSnapshot();
  });

  test('should display loading...', () => {
    setup({});
    const container = screen.queryByTestId('homeContainer');
    expect(container).toBeDefined();
    const loading = screen.queryByText(/Loading.../i);
    expect(loading).toBeDefined();
  });

  test('should load products data', async () => {
    mock.onGet('660/products').reply(200, [
      {
        id: 1,
        title: 'Fjallraven - Foldsack No. 1 Backpack, Fits 15 Laptops',
        price: 109.95,
        description:
          'Your perfect pack for everyday use and walks in the forest. Stash your laptop (up to 15 inches) in the padded sleeve, your everyday',
        category: "men's clothing",
        image: 'https://fakestoreapi.com/img/81fPKd-2AYL._AC_SL1500_.jpg',
        rating: {
          rate: 3.9,
          count: 120,
        },
      },
    ]);
    mock.onGet('660/cart').reply(200, [
      {
        productId: 1,
        quantity: 1,
        id: 2,
      },
    ]);
    setup({});
    const products = await screen.findAllByTestId('productContainer');
    expect(products.length).toBe(1);
  });

  //   test('should render products', () => {
  //     setup({
  //       products,
  //     });
  //   });
});
