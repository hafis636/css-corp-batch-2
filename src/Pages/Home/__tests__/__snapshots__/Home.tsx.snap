// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Home page cases should take snapshot 1`] = `
<div
  class="relative"
  data-testid="homeContainer"
>
  <div
    class="flex justify-center items-center text-white text-4xl w-screen h-screen bg-gray-400 absolute z-10 opacity-30"
  >
    Loading...
  </div>
</div>
`;

exports[`Test Home page cases should take snapshot of product 1`] = `
<div
  class="relative"
  data-testid="homeContainer"
>
  <div
    class="w-full relative flex items-center bg-white px-4 pt-14 pb-8 overflow-hidden shadow-2xl sm:px-6 sm:pt-8 md:p-6 lg:p-8"
    data-testid="productContainer"
  >
    <div
      class="w-full grid grid-cols-1 gap-y-8 gap-x-6 items-start sm:grid-cols-12 lg:gap-x-8"
    >
      <div
        class="aspect-w-2 aspect-h-3 rounded-lg bg-gray-100 overflow-hidden sm:col-span-4 lg:col-span-5"
      >
        <img
          alt="Fjallraven - Foldsack No. 1 Backpack, Fits 15 Laptops"
          class="object-center object-cover"
          src="https://fakestoreapi.com/img/81fPKd-2AYL._AC_SL1500_.jpg"
        />
      </div>
      <div
        class="sm:col-span-8 lg:col-span-7"
      >
        <h2
          class="text-2xl font-extrabold text-gray-900 sm:pr-12"
        >
          Fjallraven - Foldsack No. 1 Backpack, Fits 15 Laptops
        </h2>
        <section
          aria-labelledby="information-heading"
          class="mt-2"
        >
          <h3
            class="truncate"
            id="information-heading"
          >
            Your perfect pack for everyday use and walks in the forest. Stash your laptop (up to 15 inches) in the padded sleeve, your everyday
          </h3>
          <p
            class="text-2xl text-gray-900"
          >
            $109.95
          </p>
          <div
            class="mt-6"
          >
            <h4>
              Reviews
            </h4>
            <div
              class="flex items-center"
            >
              <div
                class="flex items-center"
              >
                <svg
                  aria-hidden="true"
                  class="h-5 w-5 flex-shrink-0 text-gray-900"
                  fill="currentColor"
                  viewBox="0 0 20 20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M9.049 2.927c.3-.921 1.603-.921 1.902 0l1.07 3.292a1 1 0 00.95.69h3.462c.969 0 1.371 1.24.588 1.81l-2.8 2.034a1 1 0 00-.364 1.118l1.07 3.292c.3.921-.755 1.688-1.54 1.118l-2.8-2.034a1 1 0 00-1.175 0l-2.8 2.034c-.784.57-1.838-.197-1.539-1.118l1.07-3.292a1 1 0 00-.364-1.118L2.98 8.72c-.783-.57-.38-1.81.588-1.81h3.461a1 1 0 00.951-.69l1.07-3.292z"
                  />
                </svg>
                <svg
                  aria-hidden="true"
                  class="h-5 w-5 flex-shrink-0 text-gray-900"
                  fill="currentColor"
                  viewBox="0 0 20 20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M9.049 2.927c.3-.921 1.603-.921 1.902 0l1.07 3.292a1 1 0 00.95.69h3.462c.969 0 1.371 1.24.588 1.81l-2.8 2.034a1 1 0 00-.364 1.118l1.07 3.292c.3.921-.755 1.688-1.54 1.118l-2.8-2.034a1 1 0 00-1.175 0l-2.8 2.034c-.784.57-1.838-.197-1.539-1.118l1.07-3.292a1 1 0 00-.364-1.118L2.98 8.72c-.783-.57-.38-1.81.588-1.81h3.461a1 1 0 00.951-.69l1.07-3.292z"
                  />
                </svg>
                <svg
                  aria-hidden="true"
                  class="h-5 w-5 flex-shrink-0 text-gray-900"
                  fill="currentColor"
                  viewBox="0 0 20 20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M9.049 2.927c.3-.921 1.603-.921 1.902 0l1.07 3.292a1 1 0 00.95.69h3.462c.969 0 1.371 1.24.588 1.81l-2.8 2.034a1 1 0 00-.364 1.118l1.07 3.292c.3.921-.755 1.688-1.54 1.118l-2.8-2.034a1 1 0 00-1.175 0l-2.8 2.034c-.784.57-1.838-.197-1.539-1.118l1.07-3.292a1 1 0 00-.364-1.118L2.98 8.72c-.783-.57-.38-1.81.588-1.81h3.461a1 1 0 00.951-.69l1.07-3.292z"
                  />
                </svg>
                <svg
                  aria-hidden="true"
                  class="h-5 w-5 flex-shrink-0 text-gray-900"
                  fill="currentColor"
                  viewBox="0 0 20 20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M9.049 2.927c.3-.921 1.603-.921 1.902 0l1.07 3.292a1 1 0 00.95.69h3.462c.969 0 1.371 1.24.588 1.81l-2.8 2.034a1 1 0 00-.364 1.118l1.07 3.292c.3.921-.755 1.688-1.54 1.118l-2.8-2.034a1 1 0 00-1.175 0l-2.8 2.034c-.784.57-1.838-.197-1.539-1.118l1.07-3.292a1 1 0 00-.364-1.118L2.98 8.72c-.783-.57-.38-1.81.588-1.81h3.461a1 1 0 00.951-.69l1.07-3.292z"
                  />
                </svg>
                <svg
                  aria-hidden="true"
                  class="h-5 w-5 flex-shrink-0 text-gray-900 text-gray-200"
                  fill="currentColor"
                  viewBox="0 0 20 20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M9.049 2.927c.3-.921 1.603-.921 1.902 0l1.07 3.292a1 1 0 00.95.69h3.462c.969 0 1.371 1.24.588 1.81l-2.8 2.034a1 1 0 00-.364 1.118l1.07 3.292c.3.921-.755 1.688-1.54 1.118l-2.8-2.034a1 1 0 00-1.175 0l-2.8 2.034c-.784.57-1.838-.197-1.539-1.118l1.07-3.292a1 1 0 00-.364-1.118L2.98 8.72c-.783-.57-.38-1.81.588-1.81h3.461a1 1 0 00.951-.69l1.07-3.292z"
                  />
                </svg>
              </div>
              <p
                class="sr-only"
              >
                3.9
                 out of 5 stars
              </p>
              <a
                class="ml-3 text-sm font-medium text-indigo-600 hover:text-indigo-500"
                href="#"
              >
                120
                 reviews
              </a>
            </div>
          </div>
        </section>
        <section
          aria-labelledby="options-heading"
          class="mt-10"
        >
          <h3
            id="options-heading"
          >
            men's clothing
          </h3>
          <button
            class="hover:bg-indigo-700 focus:ring-indigo-500 mt-6 w-full bg-indigo-600 border border-transparent rounded-md py-3 px-8 flex items-center justify-center text-base font-medium text-white  focus:outline-none focus:ring-2 focus:ring-offset-2 "
            type="button"
          >
            Add to bag
          </button>
        </section>
      </div>
    </div>
  </div>
  <div
    class="w-full relative flex items-center bg-white px-4 pt-14 pb-8 overflow-hidden shadow-2xl sm:px-6 sm:pt-8 md:p-6 lg:p-8"
    data-testid="productContainer"
  >
    <div
      class="w-full grid grid-cols-1 gap-y-8 gap-x-6 items-start sm:grid-cols-12 lg:gap-x-8"
    >
      <div
        class="aspect-w-2 aspect-h-3 rounded-lg bg-gray-100 overflow-hidden sm:col-span-4 lg:col-span-5"
      >
        <img
          alt="Mens Casual Premium Slim Fit T-Shirts "
          class="object-center object-cover"
          src="https://fakestoreapi.com/img/71-3HjGNDUL._AC_SY879._SX._UX._SY._UY_.jpg"
        />
      </div>
      <div
        class="sm:col-span-8 lg:col-span-7"
      >
        <h2
          class="text-2xl font-extrabold text-gray-900 sm:pr-12"
        >
          Mens Casual Premium Slim Fit T-Shirts 
        </h2>
        <section
          aria-labelledby="information-heading"
          class="mt-2"
        >
          <h3
            class="truncate"
            id="information-heading"
          >
            Slim-fitting style, contrast raglan long sleeve, three-button henley placket, light weight & soft fabric for breathable and comfortable wearing. And Solid stitched shirts with round neck made for durability and a great fit for casual fashion wear and diehard baseball fans. The Henley style round neckline includes a three-button placket.
          </h3>
          <p
            class="text-2xl text-gray-900"
          >
            $22.30
          </p>
          <div
            class="mt-6"
          >
            <h4>
              Reviews
            </h4>
            <div
              class="flex items-center"
            >
              <div
                class="flex items-center"
              >
                <svg
                  aria-hidden="true"
                  class="h-5 w-5 flex-shrink-0 text-gray-900"
                  fill="currentColor"
                  viewBox="0 0 20 20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M9.049 2.927c.3-.921 1.603-.921 1.902 0l1.07 3.292a1 1 0 00.95.69h3.462c.969 0 1.371 1.24.588 1.81l-2.8 2.034a1 1 0 00-.364 1.118l1.07 3.292c.3.921-.755 1.688-1.54 1.118l-2.8-2.034a1 1 0 00-1.175 0l-2.8 2.034c-.784.57-1.838-.197-1.539-1.118l1.07-3.292a1 1 0 00-.364-1.118L2.98 8.72c-.783-.57-.38-1.81.588-1.81h3.461a1 1 0 00.951-.69l1.07-3.292z"
                  />
                </svg>
                <svg
                  aria-hidden="true"
                  class="h-5 w-5 flex-shrink-0 text-gray-900"
                  fill="currentColor"
                  viewBox="0 0 20 20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M9.049 2.927c.3-.921 1.603-.921 1.902 0l1.07 3.292a1 1 0 00.95.69h3.462c.969 0 1.371 1.24.588 1.81l-2.8 2.034a1 1 0 00-.364 1.118l1.07 3.292c.3.921-.755 1.688-1.54 1.118l-2.8-2.034a1 1 0 00-1.175 0l-2.8 2.034c-.784.57-1.838-.197-1.539-1.118l1.07-3.292a1 1 0 00-.364-1.118L2.98 8.72c-.783-.57-.38-1.81.588-1.81h3.461a1 1 0 00.951-.69l1.07-3.292z"
                  />
                </svg>
                <svg
                  aria-hidden="true"
                  class="h-5 w-5 flex-shrink-0 text-gray-900"
                  fill="currentColor"
                  viewBox="0 0 20 20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M9.049 2.927c.3-.921 1.603-.921 1.902 0l1.07 3.292a1 1 0 00.95.69h3.462c.969 0 1.371 1.24.588 1.81l-2.8 2.034a1 1 0 00-.364 1.118l1.07 3.292c.3.921-.755 1.688-1.54 1.118l-2.8-2.034a1 1 0 00-1.175 0l-2.8 2.034c-.784.57-1.838-.197-1.539-1.118l1.07-3.292a1 1 0 00-.364-1.118L2.98 8.72c-.783-.57-.38-1.81.588-1.81h3.461a1 1 0 00.951-.69l1.07-3.292z"
                  />
                </svg>
                <svg
                  aria-hidden="true"
                  class="h-5 w-5 flex-shrink-0 text-gray-900"
                  fill="currentColor"
                  viewBox="0 0 20 20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M9.049 2.927c.3-.921 1.603-.921 1.902 0l1.07 3.292a1 1 0 00.95.69h3.462c.969 0 1.371 1.24.588 1.81l-2.8 2.034a1 1 0 00-.364 1.118l1.07 3.292c.3.921-.755 1.688-1.54 1.118l-2.8-2.034a1 1 0 00-1.175 0l-2.8 2.034c-.784.57-1.838-.197-1.539-1.118l1.07-3.292a1 1 0 00-.364-1.118L2.98 8.72c-.783-.57-.38-1.81.588-1.81h3.461a1 1 0 00.951-.69l1.07-3.292z"
                  />
                </svg>
                <svg
                  aria-hidden="true"
                  class="h-5 w-5 flex-shrink-0 text-gray-900"
                  fill="currentColor"
                  viewBox="0 0 20 20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M9.049 2.927c.3-.921 1.603-.921 1.902 0l1.07 3.292a1 1 0 00.95.69h3.462c.969 0 1.371 1.24.588 1.81l-2.8 2.034a1 1 0 00-.364 1.118l1.07 3.292c.3.921-.755 1.688-1.54 1.118l-2.8-2.034a1 1 0 00-1.175 0l-2.8 2.034c-.784.57-1.838-.197-1.539-1.118l1.07-3.292a1 1 0 00-.364-1.118L2.98 8.72c-.783-.57-.38-1.81.588-1.81h3.461a1 1 0 00.951-.69l1.07-3.292z"
                  />
                </svg>
              </div>
              <p
                class="sr-only"
              >
                4.1
                 out of 5 stars
              </p>
              <a
                class="ml-3 text-sm font-medium text-indigo-600 hover:text-indigo-500"
                href="#"
              >
                259
                 reviews
              </a>
            </div>
          </div>
        </section>
        <section
          aria-labelledby="options-heading"
          class="mt-10"
        >
          <h3
            id="options-heading"
          >
            men's clothing
          </h3>
          <div
            class="flex items-center justify-between"
            data-testid="modifyProduct"
          >
            <select
              class="mt-1 block w-1/2 py-2 px-3 border border-gray-300 bg-white disabled:bg-gray-300 rounded-md shadow-sm focus:outline-none focus:ring-indigo-500 focus:border-indigo-500 sm:text-sm"
            >
              <option
                value="1"
              >
                1
              </option>
              <option
                value="2"
              >
                2
              </option>
              <option
                value="3"
              >
                3
              </option>
              <option
                value="4"
              >
                4
              </option>
              <option
                value="5"
              >
                5
              </option>
              <option
                value="6"
              >
                6
              </option>
              <option
                value="7"
              >
                7
              </option>
              <option
                value="8"
              >
                8
              </option>
              <option
                value="9"
              >
                9
              </option>
              <option
                value="10"
              >
                10
              </option>
            </select>
            <button
              class="ml-3 text-sm font-medium text-indigo-600 hover:text-indigo-500 disabled:text-gray-600"
            >
              Delete
            </button>
          </div>
        </section>
      </div>
    </div>
  </div>
</div>
`;
